Add a new attribute - title to RESOURCE_DESCRIPTION

bostjanl-better opened the pull request.

bostjanl-better committed:
Add a new attribute `title` to RESOURCE_DESCRIPTION (SPECBASE-40)

joostholslag commented:
Suggest to update PR title (and) commit to reflect the additional change of lifecycle state type to string.

joostholslag commented:
@jelte confirmed Nedap doesn’t have an issue with this change. It is however a breaking change in both BASE and archie. There is no objection from Nedap to this change.

joostholslag commented:
@wolandscat could you please process the two changes in this branch to the UML in magic draw and update this branch with the new XMIs? Thank you.

joostholslag commented:
Related PR’s https://openehr.atlassian.net/browse/SPECPR-422
https://openehr.atlassian.net/browse/SPECPR-449

wolandscat committed:
UML changes for
SPECBASE-39 - Change `RESOURCE_DESCRIPTION._lifecycle_state_` type to `String`.
SPECBASE-40 - Add `_title_` attribute to `RESOURCE_DESCRIPTION`.

sebastian-iancu committed:
fixing description RESOURCE_DESCRIPTION.lifecycle_state according to agreement on SPECBASE-39
